### ISSUE TITLE
Lie

### ISSUE THREAD

**Issue by shacklefordjames60-coder** [mfouad/awesome-egypt-opensource#30]
# ... (Previous imports and fixtures)

def test_qec_mitigation(model):  # New
    """Test QEC reduces error impact."""
    h = torch.randn(8, model.hidden_dim)
    orig_q = model.quantum_forward_cos(h, error_rate=0.0)
    noisy_noqec = QSNNPolygraph(qec_enabled=False).quantum_forward_cos(h, error_rate=0.3)
    noisy_qec = model.quantum_forward_cos(h, error_rate=0.3)
    mse_noqec = F.mse_loss(noisy_noqec, orig_q)
    mse_qec = F.mse_loss(noisy_qec, orig_q)
    assert mse_qec < mse_noqec * 0.8, "QEC should reduce MSE by 20%+"

def test_onnx_roundtrip(model, sample_input):  # New
    """Export ONNX, load, compare inference."""
    path = export_onnx(model, sample_input[:1])
    if _HAS_ONNXRT:
        sess = ort.InferenceSession(path)
        onnx_out = sess.run(None, {'input': sample_input[:1].numpy()})[0]
        torch_out = model(sample_input[:1]).detach().numpy()
        np.testing.assert_allclose(onnx_out, torch_out, atol=1e-5)
    os.remove(path)

def test_forward_qec(model, sample_input):  # New
    """Forward pass with QEC enabled."""
    logits = model(sample_input)
    assert logits.shape == sample_input.shape[:1]

# Add to existing tests
# Run with: pytest -v --cov

if __name__ == "__main__":
    # Fixtures assumed; run manually
    device = torch.device("cpu")
    model = QSNNPolygraph(device=device, qec_enabled=True)
    sample_input = torch.randn(4, 768)
    test_qec_mitigation(model)
    test_onnx_roundtrip(model, sample_input)
    test_forward_qec(model, sample_input)
    print("v2 Tests Passed!")
